src/bun.js/bindings/bindings.cpp

```diff
@@ -2497,9 +2497,13 @@ JSC__JSValue JSC__JSValue__keys(JSC__JSGlobalObject* globalObject, JSC__JSValue
 bool JSC__JSValue__hasOwnProperty(JSC__JSValue jsValue, JSC__JSGlobalObject* globalObject, ZigString key)
 {
     JSC::VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
```

Comment:
`hasOwnProperty` should be a separate concern from `toContainKeys`. We should use `hasOwnProperty` or equiv but not make the changes within it directly.

Reply:
I think the jest implementation of `toContainKeys` calls `hasOwnProperty`

Reply:
I see now that our `hadOwnPropertyValue` is really more generic for the JSValue which is less solid around type (ex: can detect isString but not length) so is more forgiving.